packages/odyssey-react-mui/src/labs/DataComponents/DataView.tsx

```diff
@@ -0,0 +1,387 @@
+import { useFilterConversion } from "./useFilterConversion";
+import { useRowReordering } from "../../DataTable/useRowReordering";
+import {
+  DesignTokens,
+  useOdysseyDesignTokens,
+} from "../../OdysseyDesignTokensContext";
+import styled from "@emotion/styled";
+
+export type DataViewProps = UniversalProps & ViewProps<Layout>;
+
+const DataViewContainer = styled("div", {
+  shouldForwardProp: (prop) => prop !== "odysseyDesignTokens",
+})<{ odysseyDesignTokens: DesignTokens }>(({ odysseyDesignTokens }) => ({
+  display: "flex",
+  flexDirection: "column",
+  gap: odysseyDesignTokens.Spacing4,
+}));
+
+const BulkActionsContainer = styled("div")(() => ({
+  display: "flex",
+  justifyContent: "space-between",
+}));
+
+const AdditionalActionsContainer = styled("div")(() => ({
+  display: "flex",
+  justifyContent: "flex-end",
+}));
+
+const DataView = ({
+  availableLayouts = allAvailableLayouts,
+  bulkActionMenuItems,
+  currentPage = 1,
+  emptyPlaceholder,
+  errorMessage: errorMessageProp,
+  getData: getDataFn,
+  getRowId: getRowIdProp,
+  hasFilters,
+  hasPagination,
+  hasSearch,
+  hasSearchSubmitButton,
+  hasRowReordering,
+  hasRowSelection,
+  initialLayout,
+  isEmpty: isEmptyProp,
+  isLoading: isLoadingProp,
+  isNoResults: isNoResultsProp,
+  isRowReorderingDisabled,
+  noResultsPlaceholder,
+  onChangeRowSelection,
+  onReorderRows,
+  paginationType = "paged",
+  resultsPerPage = 20,
+  searchDelayTime,
+  stackOptions,
+  tableOptions,
+  totalRows,
+}: DataViewProps) => {
+  const odysseyDesignTokens = useOdysseyDesignTokens();
+
+  const [currentLayout, setCurrentLayout] = useState<Layout>(
+    initialLayout ?? availableLayouts[0],
+  );
+
+  const [data, setData] = useState<MRT_RowData[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(isLoadingProp ?? true);
+  const [isEmpty, setIsEmpty] = useState<boolean>(isEmptyProp ?? true);
+  const [isNoResults, setIsNoResults] = useState<boolean>(
+    isNoResultsProp ?? false,
+  );
+  const [errorMessage, setErrorMessage] =
+    useState<UniversalProps["errorMessage"]>(errorMessageProp);
+
+  const [search, setSearch] = useState<string>("");
+  const [initialFilters, setInitialFilters] = useState<DataFilter[]>();
+  const [filters, setFilters] = useState<DataFilter[]>();
+
+  const [draggingRow, setDraggingRow] = useState<MRT_Row<MRT_RowData> | null>();
+
+  const [rowSelection, setRowSelection] = useState<MRT_RowSelectionState>({});
+
+  useEffect(() => {
+    onChangeRowSelection?.(rowSelection);
+  }, [rowSelection, onChangeRowSelection]);
+
+  const [pagination, setPagination] = useState({
+    pageIndex: currentPage,
+    pageSize: resultsPerPage,
+  });
+
+  const [tableState, setTableState] = useState<TableState>({
+    columnSorting: [],
+    columnVisibility: {},
+    rowDensity: tableOptions?.initialDensity ?? densityValues[0],
+  });
+
+  const shouldShowFilters = useMemo(
+    () => hasSearch || hasFilters,
+    [hasSearch, hasFilters],
+  );
```

Comment:
😆 This one doesn't need to be memoized. Here's the thing, I agree, we should memoize everything and that by having everything memoized, you won't get into a situation where something isn't memoized, but this is boolean logic, and booleans are primitives. We're actually adding more compute by adding the memo in this particular use case.

If you left it in, it wouldn't make a difference performance-wise, but it looks funny regardless.